### PR TITLE
Supports the select method in the deps of apple_library

### PR DESCRIPTION
**Pull request by congt**
cc @thiagohmcruz

### CODE REVIEW

File: rules/library.bzl [hunk 1]
```diff
@@ -308,7 +308,7 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
     weak_sdk_frameworks = kwargs.pop("weak_sdk_frameworks", [])
     sdk_includes = kwargs.pop("sdk_includes", [])
     pch = kwargs.pop("pch", "@build_bazel_rules_ios//rules/library:common.pch")
-    deps = [d for d in kwargs.pop("deps", [])]
+    deps = [] + kwargs.pop("deps", [])
```
Comment (congt) [inline rules/library.bzl:311]: Got `Error: trying to mutate a frozen list value` if I use `deps = kwargs.pop("deps", [])` when building tests/framework/platforms/BUILD.bazel

File: tests/ios/frameworks/selectable-deps/BUILD.bazel [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+apple_framework(
+    name = "SelectableDepsFramework",
+    srcs = [
+        "main.m",
+    ],
+    platforms = {"ios": "10.0"},
+    visibility = ["//visibility:public"],
+    deps = select({
+        ":release": [":SelectableFramework"],
+        "//conditions:default": [":SelectableFramework"]
```
Comment (thiagohmcruz) [inline tests/ios/frameworks/selectable-deps/BUILD.bazel:32]: Missing a `,` here at the end

**Review by thiagohmcruz — APPROVED** [review]

File: tests/ios/frameworks/selectable-deps/BUILD.bazel (hunk 2 above)
Comment (thiagohmcruz) [reply tests/ios/frameworks/selectable-deps/BUILD.bazel:32]: Just noticed the unit tests are failing, never mind my previous approval we should check and see why

File: rules/library.bzl [hunk 3]
```diff
@@ -357,17 +358,18 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
...
         objc_import(
             name = import_name,
             archives = [vendored_static_library],
             tags = _MANUAL,
         )
-        deps.append(import_name)
+        vendored_deps.append(import_name)
     for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
         fail("no import for %s" % vendored_dynamic_library)
+    deps += vendored_deps
```
Comment (segiddins) [inline rules/library.bzl:372]: nice! thanks for keeping this a single array, it'll make reading `bazel query --output=build` output much nicer

**Review by segiddins — APPROVED** [review]
Awesome! I think for the sake of completeness we should also have an `ios_application` and `ios_unit_test` that have selected deps in the tests?

**Pull request merged**